R/ab_scenario.R

```diff
@@ -161,33 +180,74 @@ ab_json = function(
   if(is.null(desire_lines_out$departure)) {
     desire_lines_out$departure = time_fun(n = n, ...)
   }
-  desire_lines_out$departure = desire_lines_out$departure * 10000.0
+
+  # Do not multiply by 10k if the maximum number is already greater than 7 days
+  if(!max(desire_lines_out$departure) > 7 * 24 * 60 * 60) {
+    desire_lines_out$departure = desire_lines_out$departure * 10000
+  }
 
   start_points = lwgeom::st_startpoint(desire_lines_out) %>% sf::st_coordinates()
   end_points = lwgeom::st_endpoint(desire_lines_out) %>% sf::st_coordinates()
+  colnames(start_points) = c("ox", "oy")
+  colnames(end_points) = c("dx", "dy")
 
 
-  trips = lapply(seq(nrow(desire_lines_out)), function(i) {
-    Position_origin = data.frame(
-      longitude = start_points[i, "X"],
-      latitude = start_points[i, "Y"]
-    )
-    Position_destination = data.frame(
-      longitude = end_points[i, "X"],
-      latitude = end_points[i, "Y"]
-    )
-    origin = tibble::tibble(Position = Position_origin)
-    destination = tibble::tibble(Position = Position_destination)
-    tibble::tibble(
-      departure = desire_lines_out$departure[i],
-      origin = origin,
-      destination = destination,
-      mode = desire_lines_out[[mode_column]][i],
-      purpose = "Shopping"
+  ddf = cbind(
+    sf::st_drop_geometry(desire_lines_out),
+    start_points,
+    end_points
+  )
+  if(is.null(desire_lines_out$person)) {
+    ddf$person = seq(nrow(desire_lines_out))
+  }
+  if(is.null(desire_lines_out$purpose)) {
+    ddf$purpose = default_purpose
+  }
+  # Base R approach (tried tidyverse briefly to no avail)
+  people = lapply(unique(ddf$person), function(p) {
+    ddfp = ddf[ddf$person == p, ]
+    # trips = list(origin = list(Position = list(longitude = 1)))
+    list(
+      trips = lapply(seq(nrow(ddfp)), function(i) {
+        list(
+          departure = ddfp$departure[i],
+          origin = list(Position = list(
+            longitude = ddfp$ox[i],
+            latitude = ddfp$oy[i]
+          )),
+          destination = list(Position = list(
+            longitude = ddfp$dx[i],
+            latitude = ddfp$dy[i]
+          )),
+          mode = ddfp$mode[i],
+          purpose = ddfp$purpose[i]
+        )
+      } )
     )
-  })
+  } )
 
-  people = tibble::tibble(trips)
+  # Old approach for comparison. Todo: delete.
```

Comment:
As a reviewer, the amount of commented out code is often distracting -- especially because in github's UI, distinguishing documentation and example code from old code is hard. As long as a previous method is preserved in version control (and maybe marked with a git commit message to find it easily), how about we delete this?

Reply:
:+1: